collectors/python.d.plugin/isc_dhcpd/isc_dhcpd.chart.py

```diff
@@ -95,6 +134,30 @@ def __contains__(self, item):
         return item.address in self.network
 
 
+class PoolIPRange:
+    def __init__(self, ip_range):
+        ip_range = ip_range.split("-")
+        self.networks = list(self._summarize_address_range(ip_range[0], ip_range[1]))
```

Comment:
This is potentially very large and instantiating the list will cost at least 4-bytes + all the Python overhead per element. So for an address range in a `/8` this could be 100s of MB. The IPv4Address class support comparators so if you store the low and high values then you can test if it is in the range with `((x>=lo) and (x<=hi))` and it will be much faster and take no memory.

Reply:
> This is potentially very large and instantiating the list will cost at least 4-bytes + all the Python overhead per element.

> ((x>=lo) and (x<=hi))

yes, i use this method in `dnsmasq_dhcp` collector:
 > https://github.com/netdata/go.d.plugin/blob/5f61c42fdd4e0e4b4801d4ac574cc78a5f9a0dec/modules/dnsmasq_dhcp/ip/range.go#L90-L93

---

However i think this is fine and can be improved later. I assume ip ranges usually not huge, it is more common to use cidr.

Reply:
The CIDR case looks fine as it checks if the ip-address is inside the network range directly. If you are already handling this in the go code then it will get fixed during migration. LGTM